api/src/main/java/net/kyori/adventure/sound/SoundImpl.java

```diff
@@ -21,48 +21,54 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-package net.kyori.text.format;
+package net.kyori.adventure.sound;
 
-import net.kyori.text.Component;
-import net.kyori.text.util.NameMap;
+import java.util.stream.Stream;
+import net.kyori.adventure.key.Key;
+import net.kyori.examination.Examinable;
+import net.kyori.examination.ExaminableProperty;
 import org.checkerframework.checker.nullness.qual.NonNull;
 
-/**
- * An enumeration of colors which may be applied to a {@link Component}.
- */
-public enum TextColor implements TextFormat {
-  BLACK("black"),
-  DARK_BLUE("dark_blue"),
-  DARK_GREEN("dark_green"),
-  DARK_AQUA("dark_aqua"),
-  DARK_RED("dark_red"),
-  DARK_PURPLE("dark_purple"),
-  GOLD("gold"),
-  GRAY("gray"),
-  DARK_GRAY("dark_gray"),
-  BLUE("blue"),
-  GREEN("green"),
-  AQUA("aqua"),
-  RED("red"),
-  LIGHT_PURPLE("light_purple"),
-  YELLOW("yellow"),
-  WHITE("white");
-
-  /**
-   * The name map.
-   */
-  public static final NameMap<TextColor> NAMES = NameMap.create(TextColor.class, constant -> constant.name);
-  /**
-   * The name of this color.
-   */
-  private final String name;
+final class SoundImpl implements Examinable, Sound {
```

Comment:
Should these `Impl` classes have a `hashCode` and `equals` implementation? Might make sense, for instance, to put these in a `Map`.

Reply:
https://github.com/KyoriPowered/text/pull/62/commits/fd7ce1a567b36ac5cda35207c90354e398777f5d